### PR TITLE
ChatQnA docker gaudi quickstart

### PR DESCRIPTION
**Pull request by rofinn**
## Description

There are a lot of great detailed instruction in the ChatQnA/docker/gaudi/README.md, but it's easy to get lost when you're just trying to get started. Some examples include:

1. What should the environment variables like `http_proxy` or `host_ip` be set to?
2. What happens if I set the `host_ip` to the public IP, but I don't have a hairpin NAT setup?

Ideally, a base quickstart section should be as simple as.
```
> ./gen_env
...

> docker compose up
...
```

Now navigate to the <public_ip>:5173 in your browser.

## Issues
n/a

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [X] Others (enhancement, documentation, validation, etc.)

## Dependencies

None

## Tests

I've verified that the generated .env file works on multiple Gaudi 2 instance with a public and private IP address.
Since this is just generating a `.env` file that users can further modify I don't expect this to be breaking.

Other more breaking changes would involve modifying the compose.yaml files directly. For example, I'm not sure why `ipc: host`
is necessary, but I can make a separate PR with those changes.

### CODE REVIEW

File: ChatQnA/docker/gaudi/gen_env.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+RETRIEVER_SERVICE_HOST_IP=retriever
+RERANK_SERVICE_HOST_IP=reranking
+LLM_SERVICE_HOST_IP=llm
+
+# Used by the client UI to access backend services
+BACKEND_SERVICE_ENDPOINT="http://${public_ip}:8888/v1/chatqna"
+DATAPREP_SERVICE_ENDPOINT="http://${public_ip}:6007/v1/dataprep"
+DATAPREP_GET_FILE_ENDPOINT="http://${public_ip}:6007/v1/dataprep/get_file"
+DATAPREP_DELETE_FILE_ENDPOINT="http://${public_ip}:6007/v1/dataprep/delete_file"
+EOF
```
Comment (rofinn) [inline ChatQnA/docker/gaudi/gen_env.sh]: In relation to #739 I wonder if we could set the HABANA_VISIBLE_DEVICES as variables. My understanding is that the compose examples require two habana devices?

**Comment by kevinintel** [comment]
please fix conflicts

File: ChatQnA/docker/gaudi/gen_env.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# The internal docker endpoints addresses
+# If you were running components on different nodes you may need to specify the internal IP addresses
+TEI_EMBEDDING_ENDPOINT="http://tei-embedding-service:80"
+TEI_RERANKING_ENDPOINT="http://tei-reranking-service:80"
+TGI_LLM_ENDPOINT="http://tgi-service:80"
+REDIS_URL="redis://redis-vector-db:6379"
+INDEX_NAME="rag-redis"
+MEGA_SERVICE_HOST_IP=localhost
+EMBEDDING_SERVICE_HOST_IP=embedding
```
Comment (letonghan) [inline ChatQnA/docker/gaudi/gen_env.sh:39]: The `XXX_HOST_IP` should all be `public_ip` of the server.

File: ChatQnA/docker/gaudi/gen_env.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+EMBEDDING_MODEL_ID="BAAI/bge-base-en-v1.5"
+RERANK_MODEL_ID="BAAI/bge-reranker-base"
+LLM_MODEL_ID="Intel/neural-chat-7b-v3-3"
+
+# Huggingface token used to download the model if needed
+HUGGINGFACEHUB_API_TOKEN=${hf_api_token}
+
+# The internal docker endpoints addresses
+# If you were running components on different nodes you may need to specify the internal IP addresses
+TEI_EMBEDDING_ENDPOINT="http://tei-embedding-service:80"
```
Comment (letonghan) [inline ChatQnA/docker/gaudi/gen_env.sh:33]: The ips of all `ENDPOINT` should be `public_ip`. Why set to container name here?

File: ChatQnA/docker/gaudi/gen_env.sh (hunk 1 above)
Comment (letonghan) [inline ChatQnA/docker/gaudi/gen_env.sh:49]: Lack of the Nginx parameter settings.

File: ChatQnA/docker/gaudi/README.md [hunk 4]
```diff
@@ -2,6 +2,83 @@
...
+Navigate to this directory
+
+```bash
+> cd GenAIExamples/ChatQnA/docker/gaudi
+```
+
+Generate a default `.env` file.
+
+```bash
+> bash gen_env.sh
```
Comment (letonghan) [inline ChatQnA/docker/gaudi/README.md:22]: If you deleted the `set_env.sh` file, then all of the `get_env.sh` and `README` files should be updated at the same time for consistency.

Comment (rofinn) [reply ChatQnA/docker/gaudi/README.md:22]: Do you mean it needs to be changed in every example folder?

File: ChatQnA/docker/gaudi/gen_env.sh (hunk 3 above)
Comment (rofinn) [reply ChatQnA/docker/gaudi/gen_env.sh:33]: The container name is being used for internal docker networking here, which should be more secure and doesn't require setting up a hairpin NAT. 

https://docs.docker.com/compose/how-tos/networking/

We're still using a public IP for the endpoints accessed from the client code.

File: ChatQnA/docker/gaudi/gen_env.sh (hunk 2 above)
Comment (rofinn) [reply ChatQnA/docker/gaudi/gen_env.sh:39]: We're still using a public IP for cases that require it, like client-side variables needed for querying the server. This avoids the hairpin NAT requirement for communication between docker services.

File: ChatQnA/docker/gaudi/gen_env.sh (hunk 1 above)
Comment (rofinn) [reply ChatQnA/docker/gaudi/gen_env.sh:49]: Looks like the Nginx container is a new addition?

Comment (rofinn) [reply ChatQnA/docker/gaudi/gen_env.sh:49]: How is your Nginx container enhancing security? AFAICT, it's just exposing another unencrypted port to do a reverse proxy to services that are also publicly exposed with host networking.

**Pull request closed without merging**